content/app/github/window-helper.js

```diff
@@ -1,51 +1,129 @@
-			 */
-			getViewportClientBoundingBox : function () {
-				var width = document.documentElement.clientWidth;
-				var height = document.documentElement.clientHeight;
-				return new Box(0, 0, width, height);
-			},
-
-			/**
-			 * Scrolls vertically document body.
-			 * @param  {Number}		scrollDelta
-			 */
-			scrollTop : function(value) {
-				document.body.scrollTop = value;
-			},
+			INTERVAL_TICK : 100,	// ms
 
+		},
+
+		/**
+		 * Helper constructor.
+		 */
+		constructor : function () {
+			WindowHelper.$super.call(this);
+
+			this._selectedText = '';
+			this._intervalRef = setInterval(
+				this._onIntervalTick.bind(this),
+				WindowHelper.INTERVAL_TICK
+			);
+		},
+
+		/**
+		 * Get viewport bounding box.
+		 * @return {Box}
+		 */
+		getViewportBoundingBox : function () {
+			var width = document.documentElement.clientWidth;
+			var height = document.documentElement.clientHeight;
+			var left = document.body.scrollLeft;
+			var top = document.body.scrollTop;
+			return new Box(left, top, width, height);
+		},
+
+		/**
+		 * Get viewport bounding box relative to client visible area.
+		 * That means that viewport is at position (0,0).
+		 * @return {Box}
+		 */
+		getViewportClientBoundingBox : function () {
+			var width = document.documentElement.clientWidth;
+			var height = document.documentElement.clientHeight;
+			return new Box(0, 0, width, height);
+		},
+
+		/**
+		 * Scrolls vertically document body.
+		 * @param  {Number}		scrollDelta
+		 */
+		scrollTop : function(value) {
+			document.body.scrollTop = value;
+		},
+
+		/**
+		 * Append DOM element to page.
+		 * @param  {Element} child
+		 */
+		appendChild : function (child) {
+			document.body.appendChild(child);
+		},
+
+		/**
+		 * Remove DOM element from page.
+		 * @param  {Element} child
+		 */
+		removeChild : function (child) {
+			document.body.removeChild(child);
+		},
+
+		/**
+		 * Find ranges of provided text in page.
+		 * @param  {String} 	text
+		 * @return {Range[]}
+		 */
+		findTextRanges : function (text) {
+			return WindowHelper.findTextRangesInElement(document.body, text);
+		},
+
+		/**
+		 * Find ranges of provided text in DOM element.
+		 * @param  {Element} 	root
+		 * @param  {String} 	text
+		 * @return {Range[]}
+		 */
+		findTextRangesInElement : function (root, text) {
+			var traversing = new Traversing(root);
+			var visitor = new FindTextRangesVisitor(text);
+			traversing.traverse(visitor);
+			return visitor.ranges;
+		},
+
+		/**
+		 * Check if selected text have changed.
+		 * If it does trigger SELECT event.
+		 */
+		_checkSelectedText : function () {
+			var selection = window.getSelection();
+			var newSelectedText = selection.toString();
+			if (this._selectedText !== newSelectedText) {
+				this._selectedText = newSelectedText;
+				var evt = new Event(Event.SELECT, this._selectedText);
+				this.trigger(evt);
+			}
+		},
 
+		/**
+		 * Trigger specific actions periodically.
+		 */
+		_onIntervalTick : function () {
```

Comment:
Actually it won't work :) and there are a lot of ways to select text :(
Handing all cases requires a lot of code that will be harder for maintaining. I don't like this approach too much, but we'll probably need this for scrolling too if "display: static" is not working...
I'll live it for now, but definitely we'll need to do some optimisation for this. I can make an issue for that when I merge code.